### ISSUE TITLE
[FEATURE] Closed community for JHipster

### ISSUE THREAD

**Issue by shimopus** [jhipster/generator-jhipster#4839]
I want to propose a new feature for JHipster generator.

Now we have only one option for user creation flow - registration (or new user creation by administrator which is equal). But if I want to create closed community where new users could be only created by the administrator or somebody with special role it is impossible.

So my proposal is:
1) Add an option for generator - closed community
2) Add new role/roles which will be able to register new customers
3) Optionally - add configuration for security - selection of login methods (login form, social login, etc.)

I think that it is very frequent case and it will be good option for JHipsters

##### **Suggest a Fix**

I was made some hacks to be able to implement this functionality. It was described here http://stackoverflow.com/questions/41558190/how-to-prevent-new-users-registration-on-jhipster

##### **JHipster Version(s)**

generator-jhipster@3.9.1

**Comment by deepu105** [comment]
@shimopus sorry but this is a very specific business requirement and you should be able to achieve this by adding a new role and setting its permission accordingly. I don't think we would want to add business use cases and complicate our maintenance further. If you want to make such a feature accessible to the community you could always build it as a module for JHipster https://jhipster.github.io/modules/creating-a-module/

**Comment by shimopus** [comment]
@deepu105 Why you are calling it business use case? For me social login business use case too by this logic.

We can not add new role and permission for it as Admin role already have all permissions. But we can add an option to prevent self-registration for the generated solution. Why not?

@gmarziou, @jdubois  What do you think?

**Comment by jdubois** [comment]
I'm sorry I don't have time for this at the moment. Looks indeed like a specific business case.

**Comment by gzsombor** [comment]
I had to implement similar features, so I wouldn't say it's too far fetched. 
We've ended up implementing 'feature switches' - which is just a map of keys and values stored in the database. And added to the public ones to a  ProfileInfoResponse-like DTO, so the client could act accordingly. But obviously I haven't created a module / generator for this.

**Comment by jdubois** [comment]
OK but anyway that's a lot of work and we need to focus on JHipster 4

**Comment by deepu105** [comment]
I have tried to add role management to Jhipster long ago exactly for such
reasons and then I gave up as there wasnt enough interest. I'm really
against adding such specific functionality in the core as we already have
enough complexity to manage now. Thats why I said it could be built as a
module. Else the best way to handle such requirements is to build a dynamic
role management into JHipster and let users create and assign roles and
permissions.


Thanks & Regards,
Deepu
